### PR TITLE
📖 add Netlify instructions to releasing doc

### PR DESCRIPTION
**Pull request by ykakarap**
**What this PR does / why we need it**:

This PR adds instruction on the required actions to be performed on Netlify when we have a new release branch.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: docs/developer/releasing.md [hunk 1]
```diff
@@ -31,6 +31,14 @@ This will automatically create a PR in [k8s.io](https://github.com/kubernetes/k8
 1. Review the draft release on GitHub. Pay close attention to the `## :question: Sort these by hand` section, as it contains items that need to be manually sorted.
 1. Publish the release
 
+## Update Netlify for Cluster API book
```
Comment (CecileRobertMichon) [inline docs/developer/releasing.md:34]: should we make a note that only Netlify admins have permissions so the task is really ask an admin to do the below things?

Comment (ykakarap) [reply docs/developer/releasing.md:34]: done!

**Review by killianmuldoon — APPROVED** [review]
/lgtm

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**